bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/entity/generic/BusinessPartnerDb.kt

```diff
@@ -104,6 +105,9 @@ class BusinessPartnerDb(
     @JoinColumn(name = "address_confidence_id", unique = true)
     var addressConfidence: ConfidenceCriteriaDb?,
 
+    @Column(name = "currentness")
+    var currentness: Timestamp? = null,
```

Comment:
Suggestion from my side, since **java.time.Instant** is a more modern and efficient API for dealing with timestamps, it is advisable to switch to Instant. 
This also avoids potential issues with different time zone handling in java.sql.Timestamp.

Also, here [DbTimestamp](https://github.com/eclipse-tractusx/bpdm/blob/bedcf0703b9fa256de97d6bef2b3a0d88fce0570/bpdm-orchestrator/src/main/kotlin/org/eclipse/tractusx/bpdm/orchestrator/entity/DbTimestamp.kt#L30) wrapper for truncation can be created as we already did for orchestrator entities. This will ensure that timestamps are always truncated to microseconds, which can be useful if you want to avoid precision issues between application and the database.

Reply:
My recommendation is switch to `Instant` as is the best choice due to its modern design, better time zone handling, and higher precision. Handling null and string values can be easily managed with appropriate parsing and conversion logic.